src/wp-includes/class-wp-token-map.php

```diff
@@ -0,0 +1,751 @@
+	 *
+	 * @return WP_Token_Map|null Token map, unless unable to create it.
+	 */
+	public static function from_array( $mappings, $key_length = 2 ) {
+		$map             = new WP_Token_Map();
+		$map->key_length = $key_length;
+
+		// Start by grouping words.
+
+		$groups = array();
+		$shorts = array();
+		foreach ( $mappings as $word => $mapping ) {
+			if (
+				self::MAX_LENGTH <= strlen( $word ) ||
+				self::MAX_LENGTH <= strlen( $mapping )
+			) {
+				_doing_it_wrong(
+					__METHOD__,
+					__( 'Token Map tokens and substitutions must all be shorter than 256 bytes.' ),
+					'6.6.0'
+				);
+				return null;
+			}
+
+			$length = strlen( $word );
+
+			if ( $key_length >= $length ) {
+				$shorts[] = $word;
+			} else {
+				$group = substr( $word, 0, $key_length );
+
+				if ( ! isset( $groups[ $group ] ) ) {
+					$groups[ $group ] = array();
+				}
+
+				$groups[ $group ][] = array( substr( $word, $key_length ), $mapping );
+			}
+		}
+
+		/*
+		 * Sort the words to ensure that no smaller substring of a match masks the full match.
+		 * For example, `Cap` should not match before `CapitalDifferentialD`.
+		 */
+		usort( $shorts, 'WP_Token_Map::longest_first_then_alphabetical' );
+		foreach ( $groups as $group_key => $group ) {
+			usort(
+				$groups[ $group_key ],
+				static function ( $a, $b ) {
+					return self::longest_first_then_alphabetical( $a[0], $b[0] );
+				}
+			);
+		}
+
+		// Finally construct the optimized lookups.
+
+		foreach ( $shorts as $word ) {
+			$map->small_words     .= str_pad( $word, $key_length + 1, "\x00", STR_PAD_RIGHT );
+			$map->small_mappings[] = $mappings[ $word ];
+		}
+
+		$group_keys = array_keys( $groups );
+		sort( $group_keys );
+
+		foreach ( $group_keys as $group ) {
+			$map->groups .= "{$group}\x00";
+
+			$group_string = '';
+
+			foreach ( $groups[ $group ] as $group_word ) {
+				list( $word, $mapping ) = $group_word;
+
+				$word_length    = pack( 'C', strlen( $word ) );
+				$mapping_length = pack( 'C', strlen( $mapping ) );
+				$group_string  .= "{$word_length}{$word}{$mapping_length}{$mapping}";
+			}
+
+			$map->large_words[] = $group_string;
+		}
+
+		return $map;
+	}
+
+	/**
+	 * Creates a token map from a pre-computed table.
+	 * This skips the initialization cost of generating the table.
+	 *
+	 * This function should only be used to load data created with
+	 * WP_Token_Map::precomputed_php_source_tag().
+	 *
+	 * @since 6.6.0
+	 *
+	 * @param int    $key_length     Group key length.
+	 * @param string $groups         Group lookup index.
+	 * @param array  $large_words    Large word groups and packed strings.
+	 * @param string $small_words    Small words packed string.
+	 * @param array  $small_mappings Small word mappings.
+	 *
+	 * @return WP_Token_Map Map with precomputed data loaded.
+	 */
+	public static function from_precomputed_table( $key_length, $groups, $large_words, $small_words, $small_mappings ) {
```

Comment:
I've thought about it, and now that you bring it up, I think it would be best to eagerly include a version so that we can ignore potential conflicts. I was trying to avoid creating a separate state value too, but it may help with versioning to only have a single exported var as an array. I'll rework this.